lib/ASN1/Universal/Integer.php

```diff
@@ -42,15 +42,20 @@ public function getContent()
         return $this->value;
     }
 
+    public function rightShift($number, $positions)
+    {
+        // Shift 1 right = div / 2
+        return gmp_strval(gmp_div($number, gmp_pow(2, (int)$positions)));
+    }
+
     protected function calculateContentLength()
     {
         $nrOfOctets = 1; // we need at least one octet
-        $tmpValue = abs($this->value);
-        while ($tmpValue > 127) {
-            $tmpValue = $tmpValue >> 8;
+        $tmpValue = gmp_abs(gmp_init($this->value));
```

Comment:
Good point indeed. @afk11 can you write a test for that so we are are sure this is covered and will not come back later as a regression?